homeassistant/components/nest/__init__.py

```diff
@@ -245,3 +256,45 @@ async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
         hass.data[DOMAIN].pop(DATA_NEST_UNAVAILABLE, None)
 
     return unload_ok
+
+
+class NestEventMediaView(HomeAssistantView):
+    """Returns media for related to events for a specific device.
+
+    This is primarily used to render media for events for MediaSource. The media type
+    depends on the specific device e.g. an image, or a movie clip preview.
+    """
+
+    url = "/api/nest/event_media/{device_id}/{event_id}"
+    name = "api:nest:event_media"
+
+    def __init__(self, hass: HomeAssistant) -> None:
+        """Initialize NestEventMediaView."""
+        self.hass = hass
+
+    async def get(
+        self, request: web.Request, device_id: str, event_id: str
+    ) -> web.StreamResponse:
+        """Start a GET request."""
+        devices = await get_media_source_devices(self.hass)
```

Comment:
Let's verify the user can access the device. 

```python
user = request[KEY_HASS_USER]
entity_id = ...  # Resolve device ID to camera entity ID.
user.permissions.check_entity(entity_id, POLICY_READ)
```

(we don't have a UI for this, but if we ever do, it is good to have these checks in place)

Reply:
Should the media player do this also?

Reply:
yeah

Reply:
I did a quick look and this may require an architecture issue given the interface to the media player doesn't appear to provide this information. There was an arch discussion about a media source entity which could also be relevant.  The way the check is currently implemented is the user needs access to all entities associated with a device, to be conservative even though.in practice there is only one nest camera today) but that could lend itself well to a media player entity permission.  I'm sure though if this feature.is implemented the entire code base needs to be audited anyway, so we're just being slightly proactive here.